pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mappers/SegmentPreprocessingMapper.java

```diff
@@ -94,7 +94,7 @@ public void map(AvroKey<GenericRecord> record, NullWritable value, final Context
       if (!normalizedTimeColumnValue.equals(_sampleNormalizedTimeColumnValue)) {
```

Comment:
Can you explain on why we need to put this log? We don't enforce that a single segment need to contain only one day's of data.

Reply:
When you decide to rollup segments, we want to make sure that only one time unit is rolled up, so it is theoretically safer to fail preprocessing if more than one time unit is contained (although I know we have some segments that cross timezones, in which case there will be more than one timeunit). I wanted to fail the job entirely and not allow customers to continue, but I settled for an error.